aspnetcore/mvc/advanced/custom-model-binding/sample/CustomModelBindingSample/Binders/AuthorEntityBinder.cs

```diff
@@ -0,0 +1,75 @@
+﻿using CustomModelBindingSample.Data;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CustomModelBindingSample.Binders
+{
+    public class AuthorEntityBinder : IModelBinder
+    {
+        private readonly AppDbContext _db;
+        public AuthorEntityBinder(AppDbContext db)
+        {
+            _db = db;
+        }
+        public Task BindModelAsync(ModelBindingContext bindingContext)
+        {
+            if (bindingContext == null)
+            {
+                throw new ArgumentNullException(nameof(bindingContext));
+            }
+
+            // specify a default argument name if none is set
+            var modelName = bindingContext.ModelName;
+            if(string.IsNullOrEmpty(modelName))
+            {
+                modelName = "authorId";
+            }
```

Comment:
Ok. I can do that, but if I hard-code it then this will no longer work:
`[ModelBinder(BinderType = typeof(AuthorEntityBinder), Name ="id")]`
...and I'd like to show that you can use the attribute to specify the name.

I could have two different model types that I'm using in the demos, so for the first part use Author with its attribute, and for the second use some other type (e.g. Book) and hardcode bookId into its binder. That's a bit more work but if you feel strongly about not having the `if` check for `modelName` in the binder I can do that.